### PR TITLE
Add check for baremetal host available state

### PR DESCRIPTION
**Pull request by aneeshkp**
Follow up for BMO [PR 427 ](https://github.com/metal3-io/baremetal-operator/pull/427)
Required for the completion of BMO  [PR 340](https://github.com/metal3-io/baremetal-operator/pull/340)
In this PR, added StateAvialabel to  CAPBM  case statement
After this is merged, BMO  [PR 340](https://github.com/metal3-io/baremetal-operator/pull/340)  need to be merged and then clean up in CAPBM to remove StateReady 
Fixes [#315](https://github.com/metal3-io/baremetal-operator/issues/315)

### CODE REVIEW

File: baremetal/baremetalmachine_manager.go [hunk 1]
```diff
@@ -388,7 +388,7 @@ func (m *MachineManager) Delete(ctx context.Context) error {
 		switch host.Status.Provisioning.State {
 		case bmh.StateRegistrationError, bmh.StateRegistering,
 			bmh.StateMatchProfile, bmh.StateInspecting,
-			bmh.StateReady, bmh.StateNone:
+			bmh.StateReady, bmh.StateAvailable,bmh.StateNone:
```
Comment (zaneb) [inline baremetal/baremetalmachine_manager.go]: I'm surprised the linter didn't complain about the missing space after the comma.

File: baremetal/baremetalmachine_manager_test.go [hunk 2]
```diff
@@ -1195,6 +1195,17 @@ var _ = Describe("BareMetalMachine manager", func() {
 			Secret:              newSecret(),
 			ExpectSecretDeleted: true,
 		}),
+		Entry("Consumer ref should be removed", testCaseDelete{
```
Comment (maelk) [inline baremetal/baremetalmachine_manager_test.go]: Can you please make the entry name unique ? for example by adding the state you are testing. It helps to find a failing test.

File: controllers/baremetalmachine_controller_integration_test.go [hunk 3]
```diff
@@ -416,6 +416,42 @@ var _ = Describe("Reconcile BaremetalMachine", func() {
 			},
 		),
 		//Given: Machine has Bootstrap data available while BMMachine has no Host Annotation
+		// BMH is in available state
+		//Expected: Requeue Expected
+		//			BMHost.Spec.Image = BMmachine.Spec.Image,
+		// 			BMHost.Spec.UserData is populated
+		// 			Expect BMHost.Spec.ConsumerRef.Name = BMmachine.Name
+		Entry("Should set BMH Spec in correct state and requeue when all objects are available but no annotation",
```
Comment (maelk) [inline controllers/baremetalmachine_controller_integration_test.go]: Same here please

**Comment by maelk** [comment]
You will need to backport this PR to release-0.2 and v1alpha1 too.

**Comment by dhellmann** [comment]
> You will need to backport this PR to release-0.2 and v1alpha1 too.

We're considering closing the v1alpha1 branch (I've added an item to the community meeting agenda for a future meeting). As far as we know, only the OpenShift team was using that branch and we don't actually need it any more.

File: baremetal/baremetalmachine_manager.go (hunk 1 above)
Comment (maelk) [reply baremetal/baremetalmachine_manager.go]: Actually we have a problem with the gofmt CI job. It does not return error on formatting issues. I'll create an issue about this in the new repo.

**Comment by maelk** [comment]
@aneeshkp Can you please take care of the comments and re-open this issue agains https://github.com/metal3-io/cluster-api-provider-metal3 ? Thank you
/close

**Pull request closed without merging**